### PR TITLE
Fix UX for picking multiple values in filters

### LINKED ISSUE #40265: Copying and pasting values from a spreadsheet to a filter and insert a unique value
### Describe the bug

This is a regression from v47. We used to allow copy-paste from spreadsheets, e.g.:
image
```
<google-sheets-html-origin><style type="text/css"><!--td {border: 1px solid #cccccc;}br {mso-data-placement:same-cell;}--></style>
Gizmo
--
Gadget


```

In v47 we used to insert values like:
filter47

But now in v49 we do:
filterv49

This is a step back in ergonomics

### To Reproduce

1) create a spreadsheet with 2 values on different rows: Gizmo and Gadget
2) go to metabase product table and try filtering on the category field, see the difference

### Expected behavior

We should provide the same behavior as v47

### Logs

NA

### Information about your Metabase installation

```JSON
v49
```

### Severity

P2 (lost ergonomics)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ranquild**
Fixes https://github.com/metabase/metabase/issues/40265
Demo 1 https://www.loom.com/share/fff5821109244c8cb42b9412e04a7c95
Demo 2 https://www.loom.com/share/ba3ea79c9d684887bd79921d229bc9e2

Improvements:
- For "search" field values, the search value is automatically synced with the filter value. It means that you can type it in the input and press "Add filter"; or change the filter operator and the value will not be lost.
- We no longer show the search value in search results. The problem with the previous behavior is that it overlaps the "Add filter" button and requires an extra click when adding a filter. It also makes it hard to understand that this value is not actually from the field values.
- Allows to paste multiple value from a spreadsheet

Downsides:
- It might be harder to understand how to add a custom filter value outside of the search results if there are some search results for your value. This is now exactly like it was before - the input should lose the focus, for instance, by pressing `Tab`.

How to test:
- New -> Question -> Sample DB -> People
- Try adding a filter for People.Name
- Try pasting values from a spreadsheet like in the referenced escalation

Screenshot 2024-03-19 at 10 57 36
Screenshot 2024-03-19 at 10 58 09
Screenshot 2024-03-19 at 10 58 16

### CODE REVIEW

**Review by cdeweyx — APPROVED** [review]

File: frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx [hunk 1]
```diff
@@ -0,0 +1,130 @@
+import type { MultiSelectProps, SelectItem } from "@mantine/core";
+import { MultiSelect } from "@mantine/core";
+import { useUncontrolled } from "@mantine/hooks";
+import type { ClipboardEvent, FocusEvent } from "react";
+import { useMemo, useState } from "react";
+
+export function MultiAutocomplete({
```
Comment (ranquild) [inline frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.tsx:7]: I'll add separate unit tests in a follow-up PR. For now this is tested within `FilterValuePicker`.

**Review by nemanjaglumac — COMMENTED** [review]
Some of the things I found while testing this branch locally:

Once you paste the values, it's no longer possible to paste more values.
Kapture 2024-03-19 at 21 53 07

If you paste two values for `is` and then switch to `contains`, the second value will be lost.
Kapture 2024-03-19 at 21 55 31

Please note how "Jane Doe" appears again after I tried to paste when switching from `contains` to `is` again.

**Comment by ranquild** [comment]
@nemanjaglumac both is expected. For the first one, the reason is that we disallow creating duplicate values with copy-pasting. That's why you cannot paste the same values twice.

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx [hunk 2]
```diff
@@ -518,28 +516,27 @@ describe("StringFilterValuePicker", () => {
...
           }),
         },
       });
 
-      userEvent.type(screen.getByLabelText("Filter value"), "a@b.com");
+      userEvent.type(screen.getByLabelText("Filter value"), "a@b");
       act(() => jest.advanceTimersByTime(1000));
       expect(screen.getByText("a@b.com")).toBeInTheDocument();
-      expect(onChange).not.toHaveBeenCalled();
+      expect(screen.queryByText("a@b")).not.toBeInTheDocument();
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:539]: We can/should keep the `expect(onChange).not.toHaveBeenCalled();` assertion

File: frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+<Meta
+  title="Inputs/MultiAutocomplete"
+  component={MultiAutocomplete}
+  args={args}
+  argTypes={argTypes}
+/>
+
+# MultiAutocomplete
+
+Our themed wrapper around [Mantine MultiAutocomplete](https://v6.mantine.dev/core/multi-select/) with autocomplete features.
```
Comment (kamilmielnik) [inline frontend/src/metabase/ui/components/inputs/MultiAutocomplete/MultiAutocomplete.stories.mdx:68]:
```suggestion
Our themed wrapper around [Mantine MultiSelect](https://v6.mantine.dev/core/multi-select/) with autocomplete features.
```

File: frontend/src/metabase/querying/components/FilterValuePicker/StaticValuePicker/StaticValuePicker.tsx [hunk 4]
```diff
@@ -17,59 +16,24 @@ interface StaticValuePickerProps {
...
-    }
-  };
-
   return (
-    <MultiSelect
-      data={visibleValues}
-      value={visibleValues}
-      searchValue={searchValue}
+    <MultiAutocomplete
+      data={[]}
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/StaticValuePicker/StaticValuePicker.tsx:27]: Is this correct?

File: frontend/src/metabase/querying/components/FilterValuePicker/SearchValuePicker/SearchValuePicker.tsx [hunk 5]
```diff
@@ -2,21 +2,21 @@ import { useMemo, useState } from "react";
 import { useAsync, useDebounce } from "react-use";
 import { t } from "ttag";
 
-import { MultiSelect } from "metabase/ui";
+import { MultiAutocomplete } from "metabase/ui";
```
Comment (kamilmielnik) [inline frontend/src/metabase/querying/components/FilterValuePicker/SearchValuePicker/SearchValuePicker.tsx:5]: The issue with dragging is back.

https://github.com/metabase/metabase/assets/6830683/df1b5829-ae35-4926-a58d-37f2ff29b355

cc: @npfitz who is working on something similar

File: frontend/src/metabase/querying/components/FilterValuePicker/StaticValuePicker/StaticValuePicker.tsx (hunk 4 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/StaticValuePicker/StaticValuePicker.tsx:27]: Yes - there are no predefined options in this case

File: frontend/src/metabase/querying/components/FilterValuePicker/SearchValuePicker/SearchValuePicker.tsx (hunk 5 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/SearchValuePicker/SearchValuePicker.tsx:5]: @kamilmielnik it's not back - the branch was not up-to-date with `master`. Now it is.

File: frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx (hunk 2 above)
Comment (ranquild) [reply frontend/src/metabase/querying/components/FilterValuePicker/FilterValuePicker.unit.spec.tsx:539]: Fixed

**Review by kamilmielnik — APPROVED** [review]
Works well :+1: 

> It might be harder to understand how to add a custom filter value outside of the search results if there are some search results for your value. This is now exactly like it was before - the input should lose the focus, for instance, by pressing Tab.

Yep, definitely not ideal and it'd be nice to follow up on this with the design team.

**Pull request merged**